Makefile

```diff
@@ -2239,51 +2239,19 @@ $(BIN_DIR)/HalideTraceViz: $(ROOT_DIR)/util/HalideTraceViz.cpp $(INCLUDE_DIR)/Ha
 $(BIN_DIR)/HalideTraceDump: $(ROOT_DIR)/util/HalideTraceDump.cpp $(ROOT_DIR)/util/HalideTraceUtils.cpp $(INCLUDE_DIR)/HalideRuntime.h $(ROOT_DIR)/tools/halide_image_io.h
 	$(CXX) $(OPTIMIZE) -std=c++11 $(filter %.cpp,$^) -I$(INCLUDE_DIR) -I$(ROOT_DIR)/tools -I$(ROOT_DIR)/src/runtime -L$(BIN_DIR) $(IMAGE_IO_CXX_FLAGS) $(IMAGE_IO_LIBS) -o $@
 
-# Run clang-format on most of the source. The tutorials directory is
-# explicitly skipped, as those files are manually formatted to
-# maximize readability. NB: clang-format is *not* stable across versions;
-# we are currently standardized on the formatting from clang-format-10.
-# If CLANG_FORMAT points to a different version, you may get incorrectly-formatted code.
-CLANG_FORMAT ?= ${CLANG}-format
-
+# Note: you must have CLANG_FORMAT_LLVM_INSTALL_DIR set for this rule to work.
```

Comment:
Can we just set it to $(shell dirname $(CLANG)) here?

Reply:
clang-format isn't compatible between versions -- our current formatting is based on clang-format-10 and using other versions will produce slightly different results. So if you building with LLVM12 that would give you clang-format-12 and you'd get skewed formatting. (Yes, this is unfortunate, but the stated position of the maintainers is that they will never promise inter-version compatibility.)

Reply:
If you have the ubuntu packages clang-format-10 and clang-tidy-10 installed, this should be set to /usr/lib/llvm-10

We could use it as the default. Or maybe just add a comment to that effect?

Reply:
Sure, those seem like reasonable defaults.

Reply:
Also make variables don't become environment variables of shell scripts that you invoke. You may need CLANG_FORMAT_LLVM_INSTALL_DIR=$(CLANG_FORMAT_LLVM_INSTALL_DIR) before the shell script invocation

Reply:
Whoops.

Reply:
Seems like it's not finding the .clang-format file and then doing nothing. Investigating...

Reply:
That wasn't it - I was confusing myself. The find invocation is discovering emacs backup files and getting confused. Can we replace the find with "git ls-files *.cpp *.c *.h"?

Reply:
> Can we replace the find with "git ls-files *.cpp *.c *.h"?

Can we replace emacs with a modern editor? :-)

More seriously, `git ls-files` doesn't seem to support the `**` globstar operator, so the irregular subdir pattern in some places (eg apps) means we'll either miss some files or have to have a complex set of dirs to pass to it (which will get stale).

Does the emacs backup file setup follow a naming pattern that we can exclude from the `find`?

Reply:
Yes, but I was wondering if it was good to exclude files not in the repo, so I went with the git ls-files option first. The emacs exclude would be any file starting with a .

These args to find exclude dot files:

`\( -name *.cpp -o -name *.h -o -name *.c \) -and -not -wholename "*/.*"`

Reply:
PTAL